src/db.c

```diff
@@ -1845,7 +1837,18 @@ int getKeysFromCommandWithSpecs(struct redisCommand *cmd, robj **argv, int argc,
 
 /* This function returns a sanity check if the command may have keys. */
 int doesCommandHaveKeys(struct redisCommand *cmd) {
-    return cmd->getkeys_proc || cmd->key_specs_num;
+    return (!(cmd->flags & CMD_MODULE) && cmd->getkeys_proc) || /* has getkeys_proc (non modules) */
+        (cmd->flags & CMD_MODULE_GETKEYS) ||                    /* module with GETKEYS */
+        (getAllKeySpecsFlags(cmd, 1) & CMD_KEY_CHANNEL);       /* has at least one key-spec not marked as CHANNEL */
+}
+
+/* Returns if a given command may possibly access channels. For this context. */
+int doesCommandHaveChannels(struct redisCommand *cmd) {
+    return (getAllKeySpecsFlags(cmd, 0) & CMD_KEY_CHANNEL) || /* has at least one key-spec marked as CHANNEL */
```

Comment:
@oranagra This grabs `SUNSUBSCRIBE` as well, which we don't want. I'm going to add an explicit check for that if that works. I'm also going to test this.

Reply:
Actually, this is missing other support for modules + channels that was also removed. We can add it back, but the functionality isn't here right now.

Reply:
I'd still argue this function belongs in db.c next to the other one. 
Also, even if modules can't yet use it (so my commit comment was false), I think conceptually it's nicer to count on key spec flags (or any other flags), rather than explicitly list function names. 
I.e. The transition we're making in other places like processCommand, to stop maintaining lists of functions, and instead define their behavior in flags.